crates/stc_ts_file_analyzer/src/analyzer/expr/mod.rs

```diff
@@ -516,9 +527,17 @@ impl Analyzer<'_, '_> {
                                 }
                             }
 
-                            e.right
-                                .validate_with_args(&mut *analyzer, (mode, None, Some(&ty)))
-                                .context("tried to validate rhs an assign expr")
+                            if let Some(facts) = with_child_scope {
+                                analyzer.with_child(ScopeKind::Block, facts, |analyzer| {
```

Comment:
I think this should be something different than `ScopeKind::Block`.

Reply:
Maybe `ScopeKind::ControlFlow`?

Reply:
couldn't find `ScopeKind::ControlFlow`, maybe you meant `ScopeKind::Flow`?